### ISSUE TITLE
epoll_ctl is able to set/unset O_NONBLOCK for eventfd descriptors

### ISSUE THREAD

**Issue by shkhln** [jiixyj/epoll-shim#49]
Roughly like this:
```
% cat efd-epoll-block.c 
#include <assert.h>
#include <stdio.h>
#include <sys/eventfd.h>
#include <sys/epoll.h>

int main() {
  int fd = eventfd(0, EFD_NONBLOCK);
  assert(fd != -1);

  int epoll_fd = epoll_create(1);
  assert(epoll_fd != -1);

  struct epoll_event ee;
  ee.events   = 0;
  ee.data.u64 = 0;

  assert((fcntl(fd, F_GETFL) & O_NONBLOCK) != 0);

  printf("whatever\n");

  int err = epoll_ctl(epoll_fd, EPOLL_CTL_ADD, fd, &ee);
  assert(err == 0);

  assert((fcntl(fd, F_GETFL) & O_NONBLOCK) != 0);

  return 0;
}
% cc efd-epoll-block.c -Wall -Wextra -I/usr/local/include/libepoll-shim -L/usr/local/lib -lepoll-shim
% ./a.out
whatever
Assertion failed: ((fcntl(fd, F_GETFL) & O_NONBLOCK) != 0), function main, file efd-epoll-block.c, line 26.
Abort (core dumped)
```

This boils down to to the use of uninitialized `tmp` var at https://github.com/jiixyj/epoll-shim/blob/538cf422ee062eca456c5455f666ae5c41c3c519/src/epollfd_ctx.c#L1123

**Comment by jiixyj** [comment]
Thanks for your report! Your analysis is correct -- I adapted your test case and fixed the bug in commit 3139d763d9b5eb296532db6672bd3522866b5dc8.

Using `ioctl` and `FIONBIO` to check if a fd is a `kqueue` was not a good idea. Now, `kevent` with empty change and event lists is used. True `kqueue` fds will succeed with a return value of `0`, while non-`kqueue` fds will return an `errno` of `EBADF`.

**Comment by shkhln** [comment]
That looks quite a bit neater. Thanks!
